src/daisyworld/daisyworld/World.java

```diff
@@ -147,8 +158,11 @@ private void sproutDaisy(Patch[][] newPatches, int i, int j) {
     private void grow(Patch[][] newPatches, int i, int j, int[] coords) {
         if (!newPatches[coords[0]][coords[1]].hasDaisy() && !newPatches[coords[0]][coords[1]].justDied()) {
             double temperature = newPatches[coords[0]][coords[1]].getTemp();
+            double soilQuality = newPatches[coords[0]][coords[1]].getSoilQuality();
             // Probability check: only with temperature higher than 5 degree, 100% success at 22.5 degree
-            double rnd = 0.1457 * temperature - 0.0032 * temperature * temperature - 0.6443;
+            // Extension possibility check: rnd * soil quality, if the extension switch is off,
+            // then soil quality should be 1 and won't affect the result.
+            double rnd = (0.1457 * temperature - 0.0032 * temperature * temperature - 0.6443) * soilQuality;
```

Comment:
This reduces the overall possibility of growing new daisy by 50%. Suggest projecting the effect of soil quality into a range of 50% ~ 150%, or other ratios to make sure the expectation of growing possibility is the same

Reply:
I think that it makes sense that the soil quality would influence the growth of daisy.
When the soil quality reaches the death line, which means that this patch is turned into a piece of desert, and there will be no daisy growing on it.